pkg/controller/atlasdatabaseuser/databaseuser.go

```diff
@@ -113,27 +113,25 @@ func performUpdateInAtlas(ctx *workflow.Context, k8sClient client.Client, projec
 	retryAfterUpdate := workflow.InProgress(workflow.DatabaseUserDeploymentAppliedChanges, "Clusters are scheduled to handle database users updates")
 
 	// Try to find the user
-	u, _, err := ctx.Client.DatabaseUsers.Get(ctx.Context, dbUser.Spec.DatabaseName, project.ID(), dbUser.Spec.Username)
-	if err != nil {
-		var apiError *mongodbatlas.ErrorResponse
-		if errors.As(err, &apiError) && apiError.ErrorCode == atlas.UsernameNotFound {
-			log.Debugw("User doesn't exist. Create new user", "apiUser", apiUser)
-			if _, _, err = ctx.Client.DatabaseUsers.Create(ctx.Context, project.ID(), apiUser); err != nil {
-				return workflow.Terminate(workflow.DatabaseUserNotCreatedInAtlas, err.Error())
-			}
-			ctx.EnsureStatusOption(status.AtlasDatabaseUserPasswordVersion(currentPasswordResourceVersion))
-
-			ctx.Log.Infow("Created Atlas Database User", "name", dbUser.Spec.Username)
-			return retryAfterUpdate
-		} else {
+	au, err := dus.Get(ctx.Context, dbUser.Spec.DatabaseName, project.ID(), dbUser.Spec.Username)
```

Comment:
same comment here: we change code heavily in this file while this whole code block is not unit tested.

Reply:
This will be covered by `ensureDatabaseUser` unit tests.